src/mz-debug/src/main.rs

```diff
@@ -49,7 +49,12 @@ pub struct SelfManagedDebugMode {
     #[clap(long, default_value = "true", action = clap::ArgAction::Set)]
     dump_k8s: bool,
     /// A list of namespaces to dump.
-    #[clap(long = "k8s-namespace", required_if_eq("dump_k8s", "true"), action = clap::ArgAction::Append)]
+    #[clap(
+        long = "k8s-namespace",
+        required_unless_present = "dump_k8s",
+        required_if_eq("dump_k8s", "true"),
```

Comment:
Maybe doc these constraints (though they might be clear to someone who is more familiar so feel free to ignore)